### PR TITLE
💄 718 - Add password validation confirmation display

### PR DESCRIPTION
**Pull request by anncatton**
## Description of Changes

Adds display validation messages for Password input to support granular errors.
PR # 2 of Password changes for #718 + #726 

### Consent UI
- creates `PasswordFieldSet` component for displaying custom validation messages for a password field
- messages are intended to display once the field has been touched then blurred on validation
- adds validation pass (`Checkmark`) and fail (`ErrorX`) icons for messages
- Password input reuses text input error styling but will not display validation errors in red like regular `TextFieldSet` components
- adds translation keys for password validation messages. These keys match the error `message` values added to Zod context in the `hasRequiredPasswordElements` validation

## PR Readiness Checklist

- [x] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [x] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [ ] Tests are updated (if required) and passing
- [ ] PR feedback has been addressed **if applicable**
- [ ] New environment variables added to `.env.schema` files, `README.md`
- [x] Added copyrights to new files

### CODE REVIEW

File: apps/consent-ui/src/components/common/Form/fieldsets/FieldSet/index.tsx [hunk 1]
```diff
@@ -68,12 +68,7 @@ const FieldSet = ({
 
 			<div className={styles.inputGridArea}>
 				{children}
-
-				{Array.isArray(error) ? (
-					error.map((e) => <InputError key={e}>{e}</InputError>)
-				) : (
-					<InputError>{error}</InputError>
-				)}
+				{error && <InputError>{error}</InputError>}
```
Comment (anncatton) [inline apps/consent-ui/src/components/common/Form/fieldsets/FieldSet/index.tsx:71]: Reverting change from [earlier PR ](https://github.com/OHCRN/platform/pull/720) now that password will use a `PasswordFieldSet` component

File: apps/consent-ui/src/components/common/Form/fieldsets/PasswordFieldSet/index.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+import textFieldStyles from '../TextFieldSet/TextFieldSet.module.scss';
+import FieldSet from '../FieldSet';
+
+import styles from './PasswordFieldSet.module.scss';
+
+const messages = [
+	'missingLowercaseLetter',
+	'missingUppercaseLetter',
+	'invalidPasswordLength',
+	'invalidPasswordCharacter',
```
Comment (mistryrn) [inline apps/consent-ui/src/components/common/Form/fieldsets/PasswordFieldSet/index.tsx:40]: Should we pull these from the `ZodCustomError` enum? I don't remember if that was exported from the types package or not.

File: apps/consent-ui/src/components/common/Icons/ErrorX.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import React from 'react';
+// width="11" height="9"
```
Comment (mistryrn) [inline apps/consent-ui/src/components/common/Icons/ErrorX.tsx:21]: Should this be removed?

**Review by mistryrn — COMMENTED** [review]
More great stuff! Just a couple smol comments/questions on this one 🙂

File: apps/consent-ui/src/components/common/Form/fieldsets/PasswordFieldSet/index.tsx (hunk 2 above)
Comment (anncatton) [reply apps/consent-ui/src/components/common/Form/fieldsets/PasswordFieldSet/index.tsx:40]: i will check!

File: apps/consent-ui/src/components/common/Icons/ErrorX.tsx (hunk 3 above)
Comment (anncatton) [reply apps/consent-ui/src/components/common/Icons/ErrorX.tsx:21]: oh whoops yes 🤦‍♀️

File: apps/consent-ui/src/components/common/Form/fieldsets/PasswordFieldSet/index.tsx (hunk 2 above)
Comment (anncatton) [reply apps/consent-ui/src/components/common/Form/fieldsets/PasswordFieldSet/index.tsx:40]: yeah it wasn't exported - will add that

**Review by anncatton — COMMENTED** [review]
i will add these changes in the PR for #726

**Pull request merged**